aliyun/src/test/java/org/apache/iceberg/aliyun/oss/TestOSSFileIO.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun.oss;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClient;
+import com.aliyun.oss.OSSClientBuilder;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.io.ByteStreams;
+import org.apache.iceberg.util.SerializableSupplier;
+import org.apache.iceberg.util.SerializationUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestOSSFileIO extends AliyunOSSTestBase {
+  private static final String OSS_IMPL_CLASS = OSSFileIO.class.getName();
+
+  private final OSS ossClient = ossClient().get();
+  private final Random random = ThreadLocalRandom.current();
+  private final Configuration conf = new Configuration();
+
+  @Test
+  public void testOutputFile() throws IOException {
+    String location = randomLocation();
+    int dataSize = 1024 * 10;
+    byte[] data = randomData(dataSize);
+
+    OutputFile out = fileIO().newOutputFile(location);
+    try (OutputStream os = out.create(); InputStream is = new ByteArrayInputStream(data)) {
+      ByteStreams.copy(is, os);
+    }
```

Comment:
Could we abstract this common logic into a small method to write the memory bytes to remote oss object ? I have saw lots of the places are implementing the same logic.